File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/project/ProjectType.java
package com.esferalia.aon.occam.api.model.project;

import java.io.Serializable;

public class ProjectType  implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private Integer id;
	private Integer domain;
	private String description;
	private boolean active;
	private boolean dirty;
	
	public Integer getId() {
		return id;
	}

	public ProjectType setId(Integer id) {
		this.id = id;
		return this;
	}

	public Integer getDomain() {
		return domain;
	}

	public ProjectType setDomain(Integer domain) {
		this.domain = domain;
		return this;
	}

	public String getDescription() {
		return description;
	}

	public ProjectType setDescription(String description) {
		setDirty(true);
		this.description = description;
		return this;
	}

	public boolean isActive() {
		return active;
	}
	
	public ProjectType setActive(boolean active) {
		setDirty(true);
		this.active = active;
		return this;
	}
	
	public boolean isEmpty() {
		return getId() == null && getDomain() == null
			&& getDescription() == null;
	}

	public boolean isDirty() {
		return dirty;
	}
	
	public ProjectType setDirty(boolean dirty) {
		this.dirty = dirty;
		return this;
	}
}

File: aon-solutions/src/main/webapp/models/project/ProjectType.js
import * as LS from '../../services/localStorageService.js';

export class ProjectType {
    id;
    domain;
    description;
    active;
    dirty;

    constructor(projectType) {
        if(projectType) {
           this.id = projectType.id;
           this.domain = projectType.domain || LS.getDomainId();
           this.description = projectType.description || '';
           this.active = projectType.active || true;
           this.dirty = projectType.dirty || false;
        } else {
            this.domain = LS.getDomainId();
            this.description = '';
            this.active = true;
            this.dirty = false;
        }
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
        return this;
    }

    getDomain() {
        return this.domain;  
    }

    setDomain(domain) {
        this.domain = domain;
        return this;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.setDirty(true);
        this.description = description;
        return this;
    }

    isActive() {
        return this.active;
    }

    setActive(active){
        this.setDirty(true);
        this.active = active;
        return this;
    }

    isDirty() {
        return this.dirty;
    }

    setDirty(dirty) {
        this.dirty = dirty;
        return this;
    }
} 